monosi/tasks/base.py

```diff
@@ -0,0 +1,92 @@
+import abc
+from typing import List, Optional
+
+from monosi.anomalies import ZScoreAnomalyDetector
+from monosi.config.configuration import Configuration
+from monosi.project import Project
+
+class TaskBase:
+    def __init__(self, args, config):
+        self.args = args
+        self.config = config
+
+    @classmethod
+    def from_args(cls, args):
+        try:
+            config = Configuration.from_args(args)
+        except:
+            raise Exception("There was an issue creating the task from args.")
+
+        return cls(args, config)
+
+    @classmethod
+    def run_task(cls, *args, **kwargs):
+        task = cls(*args)
+        return task.run(*args, **kwargs)
+
+    @abc.abstractmethod
+    def run(self, *args, **kwargs):
+        raise NotImplementedError('Implementation for task does not exist.')
+
+class ProjectTask(TaskBase):
+    def __init__(self, args, config):
+        super().__init__(args, config)
+        self.project: Optional[Project] = None
+        self.task_queue: List[TaskBase] = []
```

Comment:
I wonder if there's a python type specifically for queue

Reply:
Potentially - queue is misleading naming, this really is just a list cause order is arbitrary here any way technically.. at least right now